src/header.rs

```diff
@@ -851,11 +851,10 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
       assert!(fi.cdef_bits < 4);
       self.write(2, fi.cdef_bits)?; // cdef bits
       for i in 0..(1 << fi.cdef_bits) {
-        let j = i << (3 - fi.cdef_bits);
```

Comment:
So, this has been wrong? Use (1 << fi.cdef_bits), instead of "i << (3 - fi.cdef_bits)" that is wrong.

Reply:
@ycho the problem is that this order is not what was being used when the encoder was actually applying CDEF.